src/event/block/StructureGrowEvent.php

```diff
@@ -17,13 +19,33 @@ class StructureGrowEvent extends BlockEvent implements Cancellable{
 	use CancellableTrait;
 
 	private BlockTransaction $transaction;
+	private ?Item $item;
+	private ?Player $player;
 
-	public function __construct(Block $block, BlockTransaction $transaction){
+	public function __construct(Block $block, BlockTransaction $transaction, ?Item $item, ?Player $player){
 		parent::__construct($block);
 		$this->transaction = $transaction;
+		$this->item = $item;
+		$this->player = $player;
 	}
 
 	public function getTransaction() : BlockTransaction{
 		return $this->transaction;
 	}
+
+	/**
+	 * It returns the item used from the player to grow the structure.
+	 * It returns null when the structure grows by itself.
+	 */
+	public function getItem() : ?Item{
```

Comment:
Maybe `getGrowingItem()` is better?

Reply:
I think getItem is good since it can be anything. getFertilizer might also work but be confusing in cases like bamboo on bamboo